File: 1637-widest-vertical-area-between-two-points-containing-no-points/1637-widest-vertical-area-between-two-points-containing-no-points.py
class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        arr = []
        for x in points:
            arr.append(x[0])
        arr.sort()
        ans = 0
        for i in range (len(arr)-1):
            ans = max(ans, arr[i+1]-arr[i])
        return ans

File: 1637-widest-vertical-area-between-two-points-containing-no-points/1637-widest-vertical-area-between-two-points-containing-no-points.cpp
class Solution {
public:
    Solution()
    {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    }
    
    int maxWidthOfVerticalArea(vector<vector<int>>& points) 
    {
        vector<int> x(points.size()); int ans=0;
        
        for(int i=0;i<points.size();i++)x[i]=points[i][0];
        
        sort(x.begin(),x.end());
        
        for(int i=0;i<x.size()-1;i++) ans = max(ans,x[i+1]-x[i]);
        
        return ans;
    }
};